sdk/web-pubsub/web-pubsub-client/src/webPubSubClientCredential.ts

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { GetClientAccessUrlOptions } from "./models";
+
+/**
+ * Type to be used to get clientAccessUri
+ */
+export type ClientAccessUriProvider = (options?: GetClientAccessUrlOptions) => Promise<string>;
```

Comment:
This feels like a little overkill again, if you implement this method presumably you could trivially implement `WebPubSubClientCredential` directly by naming it `getClientAccessUrl`?